src/index.ts

```diff
@@ -255,29 +361,35 @@ class WorkerInfo {
     const [[, task]] = this.taskInfos;
     return task.abortSignal !== null;
   }
+
+  currentUsage () : number {
+    if (this.isRunningAbortableTask()) return Infinity;
+    return this.taskInfos.size;
+  }
 }
 
 class ThreadPool {
   publicInterface : Piscina;
-  workers : WorkerInfo[];
+  workers : AsynchronouslyCreatedResourcePool<WorkerInfo>;
   options : FilledOptions;
   taskQueue : TaskInfo[]; // Maybe turn this into a priority queue?
   completed : number = 0;
   runTime : Histogram;
   waitTime : Histogram;
   start : number = performance.now();
   inProcessPendingMessages : boolean = false;
+  startingUp : boolean = false;
+  workerFailsDuringBootstrap : boolean = false;
 
   constructor (publicInterface : Piscina, options : Options) {
     this.publicInterface = publicInterface;
-    this.workers = [];
     this.taskQueue = [];
     this.runTime = build({ lowestDiscernibleValue: 1 });
     this.waitTime = build({ lowestDiscernibleValue: 1 });
 
     const filename =
       options.filename ? maybeFileURLToPath(options.filename) : null;
-    this.options = { ...kDefaultOptions, ...options, filename };
+    this.options = { ...kDefaultOptions, ...options, filename, maxQueue: 0 };
```

Comment:
Is forcing maxQueue to 0 here intentional?

Reply:
Yeah, it’s to make sure that `'auto'` doesn’t end up in `this.options` at any point, to comply with the changed type signature